### PR TITLE
uuid/id128 code rework

### LINKED ISSUE #3611: systemd-nspawn --volatile and machine-id
seen with systemd-230 fedora rawhide

systemd-nspawn really wants to read /etc/machine-id, but it is not present in unconfigured image:

[root@notas ~]# systemd-nspawn --volatile -M my_container
Failed to read machine ID from /var/lib/machines/my_container/etc/machine-id: No such file or directory
[root@notas ~]# systemd-nspawn --volatile --uuid=6b54f61f-797f-43ed-a835-6f439fc19788 -M my_container
Failed to read machine ID from /var/lib/machines/my_container/etc/machine-id: No such file or directory

### PR DESCRIPTION
**Pull request by poettering**
Various clean-ups, mostly in order to fix #3611.

cc @msekletar, please have a look

### CODE REVIEW

File: src/nspawn/nspawn.c [hunk 1]
```diff
@@ -2232,35 +2219,38 @@ static int mount_device(const char *what, const char *where, const char *directo
...
 
-        r = read_one_line_file(etc_machine_id, &s);
+        r = id128_read(etc_machine_id, ID128_PLAIN, &id);
         if (r < 0)
-                return log_error_errno(r, "Failed to read machine ID from %s: %m", etc_machine_id);
+                return r;
 
-        t = strstrip(s);
+        if (sd_id128_is_null(id))
+                return -EINVAL;
```
Comment (keszybz) [inline src/nspawn/nspawn.c]: ENODATA?

File: src/systemctl/systemctl.c [hunk 2]
```diff
@@ -3780,17 +3780,19 @@ static void print_status_info(
...
                 }
 
                 if (i->control_pid > 0) {
                         _cleanup_free_ char *c = NULL;
 
-                        printf(" %8s: "PID_FMT, i->main_pid ? "" : " Control", i->control_pid);
+                        if (i->main_pid > 0)
+                                fputs("; Control PID: ", stdout);
+                        else
+                                fputs("Cntrl PID: ", stdout); /* if first in column, abbreviated so it fits alignment */
```
Comment (keszybz) [inline src/systemctl/systemctl.c:3791]: Alignment with what? If with "Main PID" than this should be "Ctrl PID", no?

Comment (keszybz) [inline src/systemctl/systemctl.c:3791]: Oh, it's alignment with the whole column. In that case "Cntrl PID" is indeed fine, although "Ctrl PID" might still be more standard spelling.

**Comment by msekletar** [comment]
@poettering your nspawn changes looks fine, to me. I've also tested them on they seem to work as advertised. Thanks!

File: src/systemctl/systemctl.c (hunk 2 above)
Comment (poettering) [inline src/systemctl/systemctl.c:3791]: Hmm, but "Ctrl" sounds too much like that button on your keyboard ;-). Anyway, feel free to prep a patch that drops the "n", I'd press the green button for it!

**Pull request merged**